extension/arquillian-drone-browserstack-extension/src/main/java/org/arquillian/drone/browserstack/extension/local/BrowserStackLocalRunner.java

```diff
@@ -0,0 +1,257 @@
+        String platformBinaryNameUrl = getPlatformBinaryNameUrl();
+        File browserStackLocalZipFile =
+            new File(browserStackLocalDirectory.getPath() + File.separator + platformBinaryNameUrl);
+        String url = basicUrl + platformBinaryNameUrl;
+
+        log.info("Creating directory: " + browserStackLocalDirectory);
+        browserStackLocalDirectory.mkdir();
+
+        log.info("downloading zip file from: " + url + " to " + browserStackLocalZipFile.getPath());
+        Spacelift.task(DownloadTool.class).from(url).to(browserStackLocalZipFile.getPath()).execute().await();
+
+        log.info("extracting zip file: " + browserStackLocalZipFile + " to " + browserStackLocalDirectory.getPath());
+        Spacelift.task(browserStackLocalZipFile, UnzipTool.class).toDir(browserStackLocalDirectory.getPath()).execute()
+            .await();
+
+        log.info("marking binary file: " + browserStackLocalFile.getPath() + " as executable");
+        browserStackLocalFile.setExecutable(true);
+    }
+
+    /**
+     * Returns name of a zip file, that should contain the BrowserStackLocal binary. The name contains corresponding
+     * name of the platform the program is running on.
+     *
+     * @return Formatted name of the BrowserStackLocal zip file
+     */
+    private String getPlatformBinaryNameUrl() {
+        String binary = "BrowserStackLocal-%s.zip";
+        switch (PlatformUtils.platform().os()) {
+            case WINDOWS:
+                return String.format(binary, "win32");
+            case UNIX:
+                if (PlatformUtils.is64()) {
+                    return String.format(binary, "linux-x64");
+                } else {
+                    return String.format(binary, "linux-ia32");
+                }
+            case MACOSX:
+                return String.format(binary, "darwin-x64");
+            default:
+                throw new IllegalStateException("The current platform is not supported."
+                                                    + "Supported platforms are windows, linux and macosx."
+                                                    + "Your platform has been detected as "
+                                                    + PlatformUtils.platform().os().toString().toLowerCase() + ""
+                                                    + "from the the system property 'os.name' => '" + PlatformUtils.OS
+                                                    + "'.");
+
+        }
+    }
+
+    private boolean isEmpty(String object) {
+        return object == null || object.isEmpty();
+    }
+
+    /**
+     * This thread reads an output from the BrowserStackLocal binary and prints it on the standard output. At the same
+     * time it checks if the output contains one of the strings that indicate that the binary has been successfully
+     * started and the connection established or that another BrowserStackLocal binary is already running
+     */
+    private class Reader extends Thread {
```

Comment:
I was trying to rewrite it into Spacelift, but without any success :-(. I've discussed this code with some guys from the office and I think that it should work.